### PR TITLE
Chore: Replace vsnip with luasnip

### LINKED ISSUE #581: Replace vsnip with luasnip

### PR DESCRIPTION
**Pull request by rbrishabh**
Fixes #581

Hey @ChristianChiarulli - how does this look? Let me know if this needs any changes! Thanks

### CODE REVIEW

**Review by ChristianChiarulli — COMMENTED** [review]
It will take more than this unfortunately, I recommend testing it yourself first.

File: lua/plugins.lua [hunk 1]
```diff
@@ -53,7 +53,7 @@ return require("packer").startup(function(use)
         end
     }
 
-    use {"hrsh7th/vim-vsnip", event = "InsertEnter"}
+    use {"hrsh7th/vim-luasnip", event = "InsertEnter"}
```
Comment (jasper-schnabel) [inline lua/plugins.lua:56]:
```suggestion
    use { "L3MON4D3/LuaSnip", event = "InsertEnter" }
```

File: lua/lv-compe/init.lua [hunk 2]
```diff
@@ -79,8 +79,8 @@ end
 _G.tab_complete = function()
   if vim.fn.pumvisible() == 1 then
     return t "<C-n>"
-  elseif vim.fn.call("vsnip#available", {1}) == 1 then
-    return t "<Plug>(vsnip-expand-or-jump)"
+  elseif vim.fn.call("luasnip#available", {1}) == 1 then
+    return t "<Plug>(luasnip-expand-or-jump)"
```
Comment (jasper-schnabel) [inline lua/lv-compe/init.lua:83]:
```suggestion
    elseif require("luasnip").expand_or_jumpable() then
        return t "<Plug>luasnip-expand-or-jump"
```

File: lua/lv-compe/init.lua [hunk 3]
```diff
@@ -90,8 +90,8 @@ end
 _G.s_tab_complete = function()
   if vim.fn.pumvisible() == 1 then
     return t "<C-p>"
-  elseif vim.fn.call("vsnip#jumpable", {-1}) == 1 then
-    return t "<Plug>(vsnip-jump-prev)"
+  elseif vim.fn.call("luasnip#jumpable", {-1}) == 1 then
+    return t "<Plug>(luasnip-jump-prev)"
```
Comment (jasper-schnabel) [inline lua/lv-compe/init.lua:94]:
```suggestion
    elseif require("luasnip").jumpable(-1) then
        return t "<Plug>luasnip-jump-prev"
```

**Comment by jasper-schnabel** [comment]
It's also necessary to add
```lua
require("luasnip").config.set_config({
  history = true,
  updateevents = "InsertEnter",
})
require("luasnip/loaders/from_vscode").load()
```
as setup somewhere, with that it will work, I've tested it.

**Comment by ChristianChiarulli** [comment]
@jasper-schnabel does the plugin work after adding these changes?

**Comment by jasper-schnabel** [comment]
Yes it does, I've set it up like [here](https://github.com/jasper-schnabel/neovim/blob/master/lua/completion/init.lua).

screenshot_2021-07-06_10:52:54
screenshot_2021-07-06_10:53:02

**Comment by rafamadriz** [comment]
Luasnip doesn't work with vscode snippets that havr multiple options.

**Comment by jasper-schnabel** [comment]
Then we should probably stick to vsnip until Luasnip allows vscode snippets with  multiple options.

**Comment by abzcoding** [comment]
we should probably stick to vsnip until Luasnip allows vscode snippets with multiple options

**Pull request closed without merging**